pheap.c printf format warning

Building the _master_ branch of the SDK using GCC 14.2 with `-Wformat` option shows a format warning in `pheap.c`:

```
pico-sdk/src/common/pico_util/pheap.c:137:25: warning: format '%d' expects argument of type 'int', but argument 2 has type 'uint' {aka 'unsigned int'} [-Wformat=]
  137 |     printf("node_count %d\n", count);
      |                        ~^     ~~~~~
      |                         |     |
      |                         int   uint {aka unsigned int}
      |                        %d
```

GCC details:
```arm-none-eabi-gcc (15:14.2.rel1-1) 14.2.1 20241119```

--- a/src/common/pico_util/pheap.c
+++ b/src/common/pico_util/pheap.c
@@ -134,5 +134,5 @@ static uint ph_dump_node(pheap_t *heap, pheap_node_id_t id, void (*dump_key)(phe
 
 void ph_dump(pheap_t *heap, void (*dump_key)(pheap_node_id_t, void *), void *user_data) {
     uint count = ph_dump_node(heap, heap->root_id, dump_key, user_data, 0);
-    printf("node_count %d\n", count);
+    printf("node_count %u\n", count);
 }

